packages/pages/src/dev/server/ssr/getLocalData.ts

```diff
@@ -138,8 +154,18 @@ export const getLocalDataForSlug = async ({
   locale: string;
   slug: string;
 }) => {
-  const localData = await getLocalData((data) => {
-    return data.slug?.toString() === slug && data.locale === locale;
+  let localData: Record<string, any> | null = null;
+  await getLocalData((data) => {
+    if (data.slug && data.slug === slug && data.locale === locale) {
+      if (!localData) {
+        localData = data;
```

Comment:
this setup seems a little weird, this criterion function is meant to return a boolean for `getLocalData` but we are directly extracting the data here instead of what is returning from the actual function `getLocalData`. Is there any other we can do this to handle the error case of multiple localData files?

Reply:
ah yeah I wanted to be clever and avoid making another method (since we'd still want the other uses of getLocalData to short circuit) but I can just write another method

Reply:
The more I think about it, does it make sense to validate it here? Doesn't the CLI already throw an error if there are multiple entities with the same slug?

Reply:
I reverted back to the original getLocalData usage, but lmk if you feel there should be changes

Reply:
yes, the CLI throws an error if there are multiple entities with the same slug. However, it doesn't handle the case of multiple localData files for a single entity but from different stream definitions (e.g. if there are "stream-1" and "stream-2" both have `entityType` as location, and a single entity with slug `abington-location-123`, then the command would generate two files for each stream definition).
I believe the original getLocalData would just get whichever file it comes across first. We either need to check if the stream name in the localData filename match with what is expected for the feature page, OR throw an error if there are multiple non-static data files found.

Reply:
Ah ok that makes sense, I can update! I think the dynamic mode generate-test-data call might also need a tweak